### ISSUE TITLE
Next Kubebuilder Release v4.4.0: Planning and Alignment

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kubebuilder#4427]
### What do you want to happen?

Following the draft of the release notes for the next release (based on what has been done so far), however, see in the following comments that we have some release blockers that must be accomplished before we do the next v4.4.0

---- 

**changes since [v4.3.1](https://github.com/kubernetes-sigs/kubebuilder/releases/v4.3.1)**

## ✨ New Features

- **(kustomize/v2)**: Kubebuilder scaffolds rules under `config/rbac` to help cluster admins manage the permissions for their solutions. The comments and explanations added to these scaffolds have been improved. Furthermore, it now includes admin-specific rules to make it easier for cluster admins to work with features like aggregation (e.g., `rbac.authorization.k8s.io/aggregate-to-admin`). [More info](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles). ([#4299](https://github.com/kubernetes-sigs/kubebuilder/pull/4299))
- **(go/v4)**: Added configurations for securing Metrics Server and Prometheus integration using TLS and certificates managed by CertManager, enabling users to align their solutions with best practices, enhance security, and achieve production readiness. ([#4243](https://github.com/kubernetes-sigs/kubebuilder/pull/4243), [#4312](https://github.com/kubernetes-sigs/kubebuilder/pull/4312), [#4400](https://github.com/kubernetes-sigs/kubebuilder/pull/4400))
- **(go/v4)**: Added webhook CertWatcher and flags for custom certificate configuration [#4429](https://github.com/kubernetes-sigs/kubebuilder/pull/4429))
- **(go/v4)**: Ensures that the manager container adheres to [Restricted Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) and can operate in namespaces labelled for restricted policies, following best practices. ([#4251](https://github.com/kubernetes-sigs/kubebuilder/pull/4251)). 
- **(go/v4)**: Ensure that e2e test scaffolds will validate that the enforcement is prepared to ensure that all can run as restricted. See that the curl pod to use the metrics was updated and changed ([#4435](https://github.com/kubernetes-sigs/kubebuilder/pull/4435)) 
- **(go/v4):** Improved developer experience and maintainability by updating actions and the Makefile to utilize `go.mod` for Go version management and automating ENVTEST versioning by retrieving the version directly from `go.mod`, based on the `controller-runtime` dependency in use. ([#4385](https://github.com/kubernetes-sigs/kubebuilder/pull/4385)) ([#4401](https://github.com/kubernetes-sigs/kubebuilder/pull/4401))
- **(helm/v1-alpha):** Introduced the new `helm.kubebuilder.io/v1-alpha` plugin to enable users to distribute solutions using Helm Charts. Example scaffolds are available under [testdata/project-v4-with-plugins/dist/chart](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v4-with-plugins/dist/chart). For further information see its [documentation](link). ([#4227](https://github.com/kubernetes-sigs/kubebuilder/pull/4227), [#4315](https://github.com/kubernetes-sigs/kubebuilder/pull/4315), [#4350](https://github.com/kubernetes-sigs/kubebuilder/pull/4350), [#4351](https://github.com/kubernetes-sigs/kubebuilder/pull/4351), [#4356](https://github.com/kubernetes-sigs/kubebuilder/pull/4356), [#4357](https://github.com/kubernetes-sigs/kubebuilder/pull/4357), [#4371](https://github.com/kubernetes-sigs/kubebuilder/pull/4371), [#4377](https://github.com/kubernetes-sigs/kubebuilder/pull/4377), [#4383](https://github.com/kubernetes-sigs/kubebuilder/pull/4383), [#4380](https://github.com/kubernetes-sigs/kubebuilder/pull/4380), [#4373](https://github.com/kubernetes-sigs/kubebuilder/pull/4373), [#4386](https://github.com/kubernetes-sigs/kubebuilder/pull/4386), [#4375](https://github.com/kubernetes-sigs/kubebuilder/pull/4375), [#4388](https://github.com/kubernetes-sigs/kubebuilder/pull/4388), [#4406](https://github.com/kubernetes-sigs/kubebuilder/pull/4406), [#4399](https://github.com/kubernetes-sigs/kubebuilder/pull/4399), [#4419](https://github.com/kubernetes-sigs/kubebuilder/pull/4419),[#4451](https://github.com/kubernetes-sigs/kubebuilder/pull/4451)). _**IMPORTANT:** The `helm.kubebuilder.io/v1-alpha` is an **experimental initial version**, and further changes and improvements will be addressed in future releases._
- **(go/v4):** Added Hub and Spoke support for conversion webhooks. Developers can now scaffold webhooks for resource conversion more easily.  ([#4254](https://github.com/kubernetes-sigs/kubebuilder/pull/4254)) Example:
  ```bash
  # Create API to test conversion from v1 to v2
  $ kubebuilder create api --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
  $ kubebuilder create api --group crew --version v2 --kind FirstMate --controller=false --resource=true --make=false
  $ kubebuilder create webhook --group crew --version v1 --kind FirstMate --conversion --make=false --spoke v2
  ``` 
-  **(go/v4,kustomize/v2):** Add `app.kubernetes.io/name` label to allow more precise configurations ([#4437](https://github.com/kubernetes-sigs/kubebuilder/pull/4437))  
-  **(go/v4):** Standardize the webhook and controller test suites ([#4447](https://github.com/kubernetes-sigs/kubebuilder/pull/4447))
-  **(go/v4):** Upgrade golangci-lint from v1.61.0 to  v1.62.2 ([#4448](https://github.com/kubernetes-sigs/kubebuilder/pull/4448))
- **(go/v4):** Add support for go 1.23 ([#4446](https://github.com/kubernetes-sigs/kubebuilder/pull/4446))
- **(go/v4):** Upgraded `controller-gen` from `v0.16.4` to `v0.17.0`. ([#4254](https://github.com/kubernetes-sigs/kubebuilder/pull/4254), [#4457](https://github.com/kubernetes-sigs/kubebuilder/pull/4457))
- **(go/v4):** Add new makefile target to check and validate the linter config ([#4462](https://github.com/kubernetes-sigs/kubebuilder/pull/4462))
- **(go/v4):** Upgrade controller-runtime v0.19.1 to v0.19.4  ([#4482](https://github.com/kubernetes-sigs/kubebuilder/pull/4482))

## 🐛 Bug Fixes

- **(go/v4):** Fixed an issue where GitHub Actions workflows were overwritten by commands, ensuring users can now customize workflows freely without their changes being overridden. ([#4379](https://github.com/kubernetes-sigs/kubebuilder/pull/4379))
- **(go/v4):** Resolved linting issues in generated scaffolds. ([#4384](https://github.com/kubernetes-sigs/kubebuilder/pull/4384))
- **(kustomize/v2, go/v4):** Fixed CA injection for conversion webhooks. Previously, the CA injection patch was not accurate; The injection should occur only for CRDs, which are conversion types and not for all CRDs when a webhook with `--conversion` option is scaffolded. The issues go back to release 3.5.0 (we need to replace vars for replacements in the new customize/v2 plugin when it was introduced as alpha). It was not previously found, likely because conversion webhook features were incomplete. ([#4282](https://github.com/kubernetes-sigs/kubebuilder/pull/4282))
- **(go/v4):** Ensure that schemas are added before starting EnvTest-based suite tests for webhooks and controllers ([#4466](https://github.com/kubernetes-sigs/kubebuilder/pull/4466))
- **(go/v4):** e2e-tests: cleanup by removing calls to call make generate and manifests ([#4471](https://github.com/kubernetes-sigs/kubebuilder/pull/4471))

## What's Changed
* ✨ Adds a patch to configure ServiceMonitor to ensure TLS verification using cert-manager certificates by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4243
* ✨  Include admin ClusterRole and add it to the scaffold by @damsien in https://github.com/kubernetes-sigs/kubebuilder/pull/4299
* ✨  (go/v4): Enable `seccompProfile.type: RuntimeDefault` by default in scaffolded projects by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4251
* ✨ Add helm plugin to distribute projects by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4227
* 📖 Proposal: New Plugin to allow project distribution via helm charts by @dashanji in https://github.com/kubernetes-sigs/kubebuilder/pull/3632
* ✨ Follow-up to PR #4243: Limit permissions to access the metrics-server-cert secret by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4312
* 📖 doc: update roadmap for 2024 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4313
* ✨ (helm/v1-alpha): add missed info to chart values by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4315
* 📖 Update README.md - Fix typo by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4318
* 📖 Bump sigs.k8s.io/kubebuilder/v4 from 4.3.0 to 4.3.1 in /docs/book/src/simple-external-plugin-tutorial/testdata/sampleexternalplugin/v1 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4322
* 📖: Fix typo in CONTRIBUTING-ROLES.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4324
* :book: Fix readme wrong url your-own-plugin by @damsien in https://github.com/kubernetes-sigs/kubebuilder/pull/4325
* 📖 doc: add .github/SECURITY.md with further information by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4253
* 📖: update testdata reference by @emmanuel-ferdman in https://github.com/kubernetes-sigs/kubebuilder/pull/4328
* 📖 : Fix typo in contributing roles by @damsien in https://github.com/kubernetes-sigs/kubebuilder/pull/4329
* 🌱 : Fix `make remove-spaces` for GNU system by @damsien in https://github.com/kubernetes-sigs/kubebuilder/pull/4330
* 🌱 : Revert Fix `make remove-spaces` for GNU system" by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4331
* ✨ Upgrade controller-gen from v0.16.4 to v0.16.5 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4332
* 📖 Fix incorrect link address by @LJTian in https://github.com/kubernetes-sigs/kubebuilder/pull/4333
* 📖 Docs: removed typeDegradedMemcached and comment by @vtrenton in https://github.com/kubernetes-sigs/kubebuilder/pull/4335
* 📖 small fixes to extending_cli_features_and_plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4336
* 📖 Fix broken reference to project-file instead of project-file-config by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4337
* 📖 small fixes to extending_cli_features_and_plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4343
* 📖 fix : typo in README.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4349
* 📖 Fix: typo in external-plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4345
* 📖 small fixes to extending_cli_features_and_plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4338
* 📖 Fix: typo in external-plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4347
* 🐛 fix: (helm/v1-alpha): Remove extra space before the labels section by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4350
* 🐛 fix: (helm/alpha-v1): Fix rendered value for app.kubernetes.io/name by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4351
* 📖 Fix: missing url to plugin-utils in extending_cli_features_and_plug… by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4340
* 📖 Fix: broken urls in testing-plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4348
* 📖 update metrics documentation to bring more clarity by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4358
* 🌱 fix: run make generate to fix ci by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4353
* 📖 follow up of update metrics documentation to bring more clarity #4358 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4359
* 📖 add notice to make bring awareness to the need to move away and no longer use old images provide via the registry by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4363
* 📖 Readme: Make more clear notice about kube-rbac-proxy by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4364
* 📖 Update README.md - Fix layout from notice by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4365
* 📖 nit - remove spaces by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4366
* 🌱 Bump github.com/onsi/ginkgo/v2 from 2.21.0 to 2.22.0 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4367
* 🌱 fix file.Template to machinery.Template in the comments by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4339
* 🐛 fix: (helm/v1alpha1): add missing replicas field to the Deployment generated by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4356
* 🐛 fix: (helm/v1alpha1) add missing name prefix and namespace to leader-election-role and leader-election-rolebinding by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4357
* 🐛 fix: (helm/v1alpha1): env config should not be dependent on DeployImages by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4371
* 📖 Update discontinue_usage_of_kube_rbac_proxy.md - Typos/grammar fixes and use the right link for the new location of the images by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4374
* :seedling: Make `make remove-spaces` compatible with GNU and MAC OS X system by @damsien in https://github.com/kubernetes-sigs/kubebuilder/pull/4354
* 📖 Fix broken plugin URL in getting-started.md by @pengqun in https://github.com/kubernetes-sigs/kubebuilder/pull/4372
* 🐛 (helm/v1alpha): fix the default value of the force flag by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4375
* 📖 Fix: broken url in extending_cli_features_and_plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4341
* 📖 Fix: broken url in go-v4-plugin.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4352
* 📖 Fix: broken URL in external-plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4346
* ✨ (helm/v1alpha1): add GitHub action to make it easier to validate the chart generates by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4377
* 🐛 fix: ensure that GitHub actions are not overwritten by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4379
* 📖 Fix: broken url in extending_cli_features_and_plugins.md by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4344
* 📖 remove duplicated entry [plugin-subc], [plugin-sub-command] is the one being used by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4342
* 🐛 : (go/v4) fixes based on golangci-lint run -v by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4384
* 📖 Update roadmap_2024.md - Latest Status - Nov 2024 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4334
* 🐛 fix: (helm/v1alpha1): install the Prometheus-operator CRDs before installing a chart which configures a ServiceMonitor by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4383
* ✨ feat: (helm/v1alpha1): Allow ServiceAccount annotations to be configurable by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4380
* ✨ feat: (helm/v1alpha1): Allow extra pod labels to be configured by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4373
* ✨ (helm/v1alpha1): Add a step to render the chart in the GitHub Action by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4386
* ✨ (go/v4): Update actions to use go.mod for Go version management by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4385
* 🐛 fix: (helm/v1alpha1): Add missing prefixes in various k8s resources by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4388
* :seedling: remove everettraven from reviewers by @everettraven in https://github.com/kubernetes-sigs/kubebuilder/pull/4389
* :seedling: Bump github.com/onsi/gomega from 1.35.1 to 1.36.0 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4392
* :seedling: fix: centralize plugin key declaration by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4390
* :sparkles: (go/v4):  Add Hub and Spoke for conversion webhooks by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4254
* 🐛 fix: (helm/v1alpha1): Use the ServiceAccount name defined in the values.yaml by @wjiec in https://github.com/kubernetes-sigs/kubebuilder/pull/4406
* ✨ feat: Automate ENVTEST version from go.mod by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4401
* ✨ (deployimage/v1alpha1): Improve error handling and pointer usage for value setting in controller by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4399
* :seedling: Bump golang.org/x/tools from 0.27.0 to 0.28.0 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4409
* :book: doc: validating and parsing CRD fields by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4396
* 📖 Update getting-started.md by @dsolerh in https://github.com/kubernetes-sigs/kubebuilder/pull/4413
* :sparkles: (go/v4): add support to inject conversion webhook for the marker +kubebuilder:scaffold:e2e-webhooks-checks  by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4407
* 📖 [ISSUE-4414] Fix typo in getting-started.md by @jameskim0987 in https://github.com/kubernetes-sigs/kubebuilder/pull/4415
* 🐛 (kustomize/v2, go/v4):  Fix ca injection for conversion webhooks by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4282
* 📖 [ISSUE-4417] Fix minor hyperlink formatting in `emptymain.go` by @jameskim0987 in https://github.com/kubernetes-sigs/kubebuilder/pull/4418
* 🐛 fix: (helm/v1alpha1): remove redundant condition in if clause by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4419
* :seedling: Bump github.com/onsi/gomega from 1.36.0 to 1.36.1 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4421
* 🌱Build the kubebuilder binary before and use it to regenerate the helm chart by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4420
* ✨ feat: add webhook CertWatcher and flags for custom certificate configuration by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4429
* ✨ (go/v4): feat/fix: enhance cert-manager integration for metrics endpoints (follow-up to PR #4243) by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4400
* 🌱 : Ensure that golangci-lint CI version is passed for the templates by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4432
* 📖 Fix: move v2vsv3.md to the legacy folder and update references by @monteiro-renato in https://github.com/kubernetes-sigs/kubebuilder/pull/4431
* ✨ feat: enforce restricted Pod Security Context Compliance in testing by @lunarwhite in https://github.com/kubernetes-sigs/kubebuilder/pull/4435
* 🌱 Bump github.com/onsi/ginkgo/v2 from 2.22.0 to 2.22.1 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4436
* 🌱  add better description to MarshalYAML and UnmarshalYAML methods by @fengshunli in https://github.com/kubernetes-sigs/kubebuilder/pull/4438
* ✨ (kustomize/v2) feat: add app.kubernetes.io/name label to allow more precise configurations  by @wjiec in https://github.com/kubernetes-sigs/kubebuilder/pull/4437
* 🌱 Bump github.com/onsi/gomega from 1.36.1 to 1.36.2 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4440
*  📖improve tutorial test docs by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4443
* 🌱improve Ginkgo/Gomega test style by @Sijoma in https://github.com/kubernetes-sigs/kubebuilder/pull/4426
* 🌱 (ci): Improve GitHub Action to linter samples under testdata by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4444
* ✨ (go/v4): Standardize the webhook and controller test suites by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4447
* ✨ Upgrade golangci-lint from v1.61.0 to  v1.62.2 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4448
* 📖 (doc): Add "Kubebuilder Project Roadmap 2025" by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4316
* ✨ Upgrade scaffolds to use go 1.23 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4446
* 🌱 e2e - ensure that error from command is checked by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4452
* :seedling: Bump github.com/onsi/ginkgo/v2 from 2.22.1 to 2.22.2 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4455
* :seedling: fix: use Go version from go.mod file in CI by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4454
* 🌱 e2e tests improve assertions by simplify style of checks by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4456
* ✨ Upgrade controller-gen from v0.16.5 to v0.17.0 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4457
* 🌱 Add lint-config target to validate config and fix issues faced by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4458
* 🌱 Enable lint check empty-block and fix issues by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4459
* 🌱 Enable var-naming lint and fix issues for Kubebuilder CLI by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4460
* 🌱 Enable if-return linter check and fix issues for Kubebuilder CLI by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4461
* ✨ (go/v4): Add new makefile target to check and validate the linter config by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4462
* 🌱 Enable the lint to check exported methods and fix issues by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4463
* 🐛 (go/v4) Ensure that schemas are added before starting EnvTest-based suite tests for webhooks and controllers by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4466
* 🌱 Enable govet settings and fix the issue found by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4465
* 🌱 simplify E2E Book Samples job by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4469
* 🌱 fix: remaining Go upgrades by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4470
* 🌱 fix: remove unused nolint comments by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4468
* 🌱 Bump golang.org/x/tools from 0.28.0 to 0.29.0 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4473
* 🐛  e2e-tests: cleanup by removing calls to call make generate and manifests by @mateusoliveira43 in https://github.com/kubernetes-sigs/kubebuilder/pull/4471
* 🌱 Bump github.com/spf13/afero from 1.11.0 to 1.12.0 by @dependabot in https://github.com/kubernetes-sigs/kubebuilder/pull/4480
* ✨ Upgrade controller-runtime v0.19.1 to v0.19.4 by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4482
* 🐛 (helm/v1alpha1) - fix webhooks generation by removing data from helm chart values by @camilamacedo86 in https://github.com/kubernetes-sigs/kubebuilder/pull/4451

## New Contributors
* @emmanuel-ferdman made their first contribution in https://github.com/kubernetes-sigs/kubebuilder/pull/4328
* @LJTian made their first contribution in https://github.com/kubernetes-sigs/kubebuilder/pull/4333
* @pengqun made their first contribution in https://github.com/kubernetes-sigs/kubebuilder/pull/4372
* @wjiec made their first contribution in https://github.com/kubernetes-sigs/kubebuilder/pull/4406
* @dsolerh made their first contribution in https://github.com/kubernetes-sigs/kubebuilder/pull/4413
* @jameskim0987 made their first contribution in https://github.com/kubernetes-sigs/kubebuilder/pull/4415

**Full Changelog**: https://github.com/kubernetes-sigs/kubebuilder/compare/v4.3.1...master

--- Draft

**Comment by camilamacedo86** [comment]
Following the current release blockers:
- Merge the changes of https://github.com/kubernetes-sigs/kubebuilder/pull/4400 for the new feature
- We cannot expose the data from webhooks in the chart values, ideally, we should change it prior to the release OR remove the support for webhooks, see: https://github.com/kubernetes-sigs/kubebuilder/issues/4382

Nice to have:
- https://github.com/kubernetes-sigs/kubebuilder/issues/4398

**Comment by camilamacedo86** [comment]
For we can do the release, we need to:
- Get the change https://github.com/kubernetes-sigs/kubebuilder/pull/4451 merged before we release the first version of the chart
- Bump controller-runtime and controller-tools to add support for k8s 1.32

**Comment by camilamacedo86** [comment]
It seems that this issue currently blocks the Controller + Runtime (C+R) release for v0.20: [kubernetes-sigs/controller-runtime#3055](https://github.com/kubernetes-sigs/controller-runtime/issues/3055).

Since this might take a while to resolve, I plan to proceed this weekend with a release that includes the latest patch,, 0.19.4. We have a lot of changes to deliver, so this seems like the best course of action for now.

The downside is that we may need to release another version soon after this one, incorporating v0.20 but likely without significant additional changes.

If someone has any objection, please feel free to let me know. 

c/c @varshaprasad96

**Comment by camilamacedo86** [comment]
The release was done. We are preparing for the next one with k8s bumps since controller-runtime v0.20 was released.
